torch/cuda/amp/grad_scaler.py

```diff
@@ -1,160 +1,38 @@
-from collections import defaultdict, abc
-from enum import Enum
-from typing import Any, Dict, List, Optional, Tuple, cast
-import inspect
+from collections import abc
+from typing import List
 import warnings
 
 import torch
 from .common import amp_definitely_not_available
 
 
-__all__ = ["OptState", "GradScaler"]
-
-class _MultiDeviceReplicator:
-    """
-    Lazily serves copies of a tensor to requested devices.  Copies are cached per-device.
-    """
-    def __init__(self, master_tensor: torch.Tensor) -> None:
-        assert master_tensor.is_cuda or master_tensor.device.type == 'xla'
-        self.master = master_tensor
-        self._per_device_tensors: Dict[torch.device, torch.Tensor] = {}
-
-    def get(self, device) -> torch.Tensor:
-        retval = self._per_device_tensors.get(device, None)
-        if retval is None:
-            retval = self.master.to(device=device, non_blocking=True, copy=True)
-            self._per_device_tensors[device] = retval
-        return retval
-
-
-# Defines default_factory for GradScaler's _per_optimizer_states defaultdict,
-# as well as associated "enum" values.  Prefers defining these at top level because
-# - Lambdas can't be pickled, so we don't want to supply a lambda as the factory.
-# - Defining READY, UNSCALED, STEPPED and _refresh_per_optimizer_state within GradScaler
-#   causes a circular reference, which we'd rather avoid.
-class OptState(Enum):
-    READY = 0
-    UNSCALED = 1
-    STEPPED = 2
-
+__all__ = ["GradScaler"]
 
 def _refresh_per_optimizer_state():
-    return {"stage": OptState.READY, "found_inf_per_device": {}}
-
-
-class GradScaler:
-    _scale: Optional[torch.Tensor]
-    _grows_tracker: Optional[torch.Tensor]
-    _per_optimizer_states: Dict[int, Dict[str, Any]]
+    message = ("torch.cuda.amp.grad_scaler._refresh_per_optimizer_state "
+               "is deprecated. It will be removed in the future and "
+               "use torch.amp.grad_scaler._refresh_per_optimizer_state()")
```

Comment:
```suggestion
               "is deprecated. It will be removed in the future. Instead "
               "use torch.amp.grad_scaler._refresh_per_optimizer_state()")
```
nit for clarity